packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/queued_item_store/queued_item_store.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+/// Database for storing strings
+abstract class QueuedItemStore {
+  /// Insert an item to the end of the queue.
+  FutureOr<void> add(String string);
+
+  /// Get the first [count] items from the queue.
+  FutureOr<Iterable<QueuedItem>> get(int count);
+
+  /// Delete [items] from the store.
+  FutureOr<void> deleteItems(Iterable<QueuedItem> items);
+
+  /// Clear the queue of items
+  FutureOr<void> clear();
+
+  /// Closes this store and release resources
+  FutureOr<void> close();
```

Comment:
Do these need to be `FutureOr` instead of `Future`?

Reply:
Made it `FutureOr` because of `InMemoryQueuedItemStore`